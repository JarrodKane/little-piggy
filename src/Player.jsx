import React from "react";
import styled from "styled-components";

//TODO - Reduce these two styles so they are not both the same repeated code
const PlayerActive = styled.div`
  background: rgba(255, 255, 204, 0.5);
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 30%;
  border-radius: 25px;
  align-items: center;
  justify-content: center;
`;

const PlayerInactive = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 30%;
  border-radius: 25px;
  align-items: center;
  justify-content: center;
`;

const Player = (props) => {
  const { player, hasWon, score, active } = props;

  if (active === player) {
    return (
      <PlayerActive>
        <h2>Player: {player}</h2>
        <h1>{score}</h1>
      </PlayerActive>
    );
  } else {
    return (
      <PlayerInactive>
        <h2>Player: {player}</h2>
        <h1>{score}</h1>
      </PlayerInactive>
    );
  }
};

export default Player;
