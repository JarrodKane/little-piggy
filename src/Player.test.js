import React from 'react';
import { render } from '@testing-library/react';

import Player from './Player';

test('<Player>', () => {
  const { container } = render(<Player />);
  //snapshot testing
  expect(container.firstChild).toMatchSnapshot();
});
