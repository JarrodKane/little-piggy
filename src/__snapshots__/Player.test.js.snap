// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player> 1`] = `
<div
  class="sc-AxjAm kNroUf"
>
  <h2>
    Player: 
  </h2>
  <h1 />
</div>
`;
